addons/stock/models/stock_location.py

```diff
@@ -291,8 +294,12 @@ def _check_product_storage(self, product, quantity, location_qty):
             # check if only allow same product
             if self.storage_category_id.allow_new_product == "same" and self.quant_ids and self.quant_ids.product_id != product:
                 return False
+
             # check if enough space
             product_capacity = self.storage_category_id.product_capacity_ids.filtered(lambda pc: pc.product_id == product)
+            # To handle new line without quantity in order to avoid suggesting a location already full
+            if product_capacity and location_qty >= product_capacity.quantity:
+                return False
```

Comment:
If I understand correctly, it only prevent odoo suggest a loction that is completely full when creating a new line. It still doesn't take the unsaved lines into consideration. right?

Reply:
I think this part is not very necessary. Although we suggested "shelf 1" when qty is 0, but after the qty changed, we should give a correct suggestion. The reason why it doesn't give a correct suggestion after the qty changed is that the dest_location is not "shelf 1", when we try to find a putaway location, I search for a child location of "shelf 1". We should do that for the original dest_location. I fixed it in https://github.com/odoo/odoo/pull/71540/commits/03fc3fc8b99824bf40dc86a70a2ac12fb242c5a6

Reply:
Yep the idea is just to avoid suggesting a location completely full even if it will be update with the quantity done just after. Did you fix that?

Reply:
okk, then I think it's fine